src/st-flash/flash.h

```diff
@@ -11,7 +11,7 @@
 
 enum flash_cmd {FLASH_CMD_NONE = 0, FLASH_CMD_WRITE = 1, FLASH_CMD_READ = 2, FLASH_CMD_ERASE = 3, CMD_RESET = 4};
 enum flash_format {FLASH_FORMAT_BINARY = 0, FLASH_FORMAT_IHEX = 1};
-enum flash_area {FLASH_MAIN_MEMORY = 0, FLASH_SYSTEM_MEMORY = 1, FLASH_OTP = 2, FLASH_OPTION_BYTES = 3};
+enum flash_area {FLASH_MAIN_MEMORY = 0, FLASH_SYSTEM_MEMORY = 1, FLASH_OTP = 2, FLASH_OPTION_BYTES = 3, FLASH_OPTION_BYTES_BOOT_ADD = 4, FLASH_OPTCR = 5, FLASH_OPTCR1 = 6};
```

Comment:
These enum value names (flash_optcr and optcr1) using registers target register names can be really confusing and likely to clash with actual target register defines.

Reply:
I'm not really sure what you mean with the names being confusing and likely to clash.

The reason I added OPTCR and OPTCR1, is becase that is the actual name of the registers, so when you have the datasheet at hand, the OPTCR and OPTCR1 is what you see you need to use.
The FLASH_OPTION_BYTES_BOOT_ADD is the functional purpose of those specific option bytes. In the ST-LINK flash GUI you can set them directly using a name like that, so that made the most sense to me here too.

Reply:
If possible, we should align with the official documentation here.

Reply:
I'm not sure what you would want to change. The OPTCR and OPTCR1 are part of the official ST documentation and the ST-LINK flash GUI let's you wrote the BOOT_ADD bytes directly. So that's what I had implemented

Reply:
Nothing, as long as your proposal for the enum value names matches the ones in the datasheet, everything should be fine. You may resolve this conversation.